### ISSUE TITLE
Newly populated private volumes don’t get proper SELinux labels

### ISSUE THREAD

**Issue by SiGraphics** [QubesOS/qubes-issues#8242]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release

R4.2.0-rc1 (and most recent 4.2 weekly build)

### Brief summary

_[sys-net] Connection Failure_

**Failed to add/activate connection**

_failure adding connection: error writing to file ‘/etc/NetworkManager/system-connections/{wireless name}.nmconnection’: failed to create file /etc/NetworkManager/system-connections/{wireless name}.nmconnection.46E551: Permission denied_

### Steps to reproduce

Install R4.2.0-rc1
Attempt to use WiFi

### Expected behavior

WiFi works

### Actual behavior

WiFi doesn't work

Don’t have access to a LAN connection, so unable to test.

**Comment by jameycorreia** [comment]
This seems to be caused by selinux contexts not persisting in /rw when sys-net is based on a Fedora 38 disposable. A simple `sudo restorecon -vrF /rw` in sys-net resolves it in my testing. It appears that a sys-net based on non-disposable fedora-38 or a disposable debian-11 isn't affected.

**Comment by tlaurion** [comment]
@marmarek so how to fix this on fedora38 based templates so disp sys-net can be used

**Comment by DemiMarie** [comment]
@tlaurion Looks like a missing relabel after populating `/rw`.

**Comment by tlaurion** [comment]
@DemiMarie still not fixed in weekly iso?

**Comment by tlaurion** [comment]
> It appears that a sys-net based on non-disposable fedora-38 or a disposable debian-11 isn't affected.

Ditching fedora for debian for service dispvms for now to test further Q4.2 without the annoyances on weekly isos.

**Comment by tlaurion** [comment]
@marmarek 

https://openqa.qubes-os.org/tests/80558/asset/iso/Qubes-4.2.202308242042-x86_64.iso

Still affected :(

**Comment by tlaurion** [comment]
@marmarek 

https://openqa.qubes-os.org/tests/80558/asset/iso/Qubes-4.2.202308242042-x86_64.iso

Still affected :(

![signal-2023-08-25-10-26-27-233.jpg](https://github.com/QubesOS/qubes-issues/assets/827570/691459a7-3ab6-4615-8bc2-2b6d3e8b4bff)

**Comment by tlaurion** [comment]
@andrewdavidwong package is in the testing repo but not in template. This is a corner case of "please update" that is not possible easily.


We can conclude that testers are not using dispvm sys-net over fedora still today.

**Comment by andrewdavidwong** [comment]
> We can conclude that testers are not using dispvm sys-net over fedora still today.

Well, it looks like this was also reported in https://github.com/QubesOS/qubes-issues/issues/8445, so perhaps that is not entirely true.

**Comment by DemiMarie** [comment]
> > We can conclude that testers are not using dispvm sys-net over fedora still today.
> 
> Well, it looks like this was also reported in #8445, so perhaps that is not entirely true.

This problem breaks updates, so it needs to be fixed via a new ISO and a new template image.  @marmarek can you make sure that updated templates are available before rc2?

**Comment by andrewdavidwong** [comment]
> > > We can conclude that testers are not using dispvm sys-net over fedora still today.
> > 
> > 
> > Well, it looks like this was also reported in #8445, so perhaps that is not entirely true.
> 
> This problem breaks updates, so it needs to be fixed via a new ISO and a new template image. @marmarek can you make sure that updated templates are available before rc2?

I think there is a misunderstanding. I took @tlaurion to be saying something like, "This is still broken in the latest ISO, yet no one else has reported it. Therefore, we can conclude that no one else has tested this particular thing." I pointed out that at least one other person appears to have tested it, namely @SvenSemmler, as evidenced by the fact that he appears to have reported the same bug in #8445.

**Comment by tlaurion** [comment]
> > > > We can conclude that testers are not using dispvm sys-net over fedora still today.
> > > 
> > > 
> > > Well, it looks like this was also reported in #8445, so perhaps that is not entirely true.
> > 
> > This problem breaks updates, so it needs to be fixed via a new ISO and a new template image. @marmarek can you make sure that updated templates are available before rc2?
> 
> I think there is a misunderstanding. I took @tlaurion to be saying something like, "This is still broken in the latest ISO, yet no one else has reported it. Therefore, we can conclude that no one else has tested this particular thing." I pointed out that at least one other person appears to have tested it, namely @SvenSemmler, as evidenced by the fact that he appears to have reported the same bug in #8445.


I thank @SvenSemmler personally for having tested Heads latest t430-maximized changes and having through testing q4.2 rc1 for which this bug report preceded, showing that openqa doesn't completely does its job (testing installer's proposed options, or at least the defaults (fedora template) + customizations (including dispvm service VMs, including sys-net) which should have seen the regression prior of any release.

What I'm saying here is that testing processes have issues, and it's difficult without test plan to know if things work. If things are not working, just like a bad knee that becomes the norm, any issue walking becomes the norm and we rely on crotches, or mitigations, to not suffer of the problem.

The issue here was closed. Closed because the issue was thought fixed. Testers would search for open issues and don't find it anymore, and then have reasonable expectations that the last release would not contain a closed bug. Testers the same with either openqa ISO's which I learned today talking to Marek directly, do no contain updated templates so do not contain the fixe.

This cascades down to caching issues of CI, so that templates can be triggered to be rebuilt when something impacting installation is fixed. This is clearly not the case here.

And weekly ISO's are also expected to contain the updated templates, which is clearly not the case here. Sidenote, the last 'weekly' iso is from August 8th, I do not know what's up with that.

Issues are not the place to discuss those things, @andrewdavidwong you reminded me many times in the past.

My question, for the quality of quebesos which relies on tests and testers, just like Heads would be: where and how to do that properly.

I have created a thread tagging @marmarek on the forum named testing what where and how. I would suggest that the processes of openqa and weekly ISO's builds are clarified, as well as when an issue is fixed even though templates are not updated. My point still stands. That issue still not being fixed in installer shows dispvm sys-net on top of fedora is not tested otherwise many much more reports would have flowed here. I stopped testing q4.2 because of encountered and reported issues. I saw the issue closed and was willing to test to realize it's not fixed. To now understand that packages are updated but not the tempates packages and not the openqa ISO's nor weekly ISO's are built anymore?

Can we discuss about those problems in the forum maybe? The need is to be able to expect something. I remember the support load on my side when qubes 4.1 went out from end users. I hope q4.2 will be different, but I know it depends on testers. So how do we organize and make the processes known to the testers?

Thank you

**Comment by Augsch123** [comment]
I'm using disposable sys-net. But I have a different setup. I have a sys-net-dvm which is an app qube as the disposable template, and a sys-net which is a disposable. 

I also got the same error as shown in tlaurion's screenshot when I had changed my sys-net-dvm's template qube to fedora-38 and was trying to boot up sys-net. I guessed this had to do with disposable, so I booted up sys-net-dvm and connected to wifi there. Then I shut down sys-net-dvm and booted sys-net, and the error went away, so I didn't report this to the issue tracker.

**Comment by andrewdavidwong** [comment]
> The issue here was closed. Closed because the issue was thought fixed. Testers would search for open issues and don't find it anymore, and then have reasonable expectations that the last release would not contain a closed bug.

Sometimes, the devs believe they have fixed a bug, but it turns out that it was not really fixed. In that case, we reopen the issue, and the developers make another attempt at fixing it. This is just part of the reality of software development. Humans are imperfect and make mistakes. There is no way to guarantee an error-free software development process as long as humans are involved. Also, we are still in the testing stage of this release, so there should be an even greater expectation that mistakes will be made and corrected.

Closing an issue is, in effect, nothing more than adding a virtual tag on the issue to represent a certain state. Issues can be opened and closed infinitely with a single button at no cost. Do not think of an issue being closed as meaning more than it does. It certainly is not "final" in any sense. We [explicitly instruct](https://www.qubes-os.org/doc/issue-tracking/) all users of the issue tracker to search **both open and closed** issues. All the "open" and "closed" states really mean are "according to our available information at the present time, there is something actionable here" or "according to our available information at the present time, there is nothing actionable here," respectively. Since new information is constantly coming in, these states can change without warning at any moment.

> Issues are not the place to discuss those things, @andrewdavidwong you reminded me many times in the past.

Correct. This is not the right place for this discussion.

> My question, for the quality of quebesos which relies on tests and testers, just like Heads would be: where and how to do that properly.

There is a special section of the forum for [testing](https://forum.qubes-os.org/c/testing-team/28). There is also [qubes-devel](https://www.qubes-os.org/support/#qubes-devel).

**Comment by DemiMarie** [comment]
Does this problem impact anyone whose ISO includes https://github.com/QubesOS/qubes-core-agent-linux/pull/437?

**Comment by tlaurion** [comment]
> Does this problem impact anyone whose ISO includes https://github.com/QubesOS/qubes-core-agent-linux/pull/437?

@DemiMarie What ISO are you referring to? Short version: This issue was closed but not fixed in templates or ISOs. Testers expect ISOs last to work. The openqa ISOs and the weekly ISOs do not have updated templates containing updated package. The last ‘weekly’ iso is from August 8th. Today's openqa build might contain the fix but I'm not sure.

TLDR: Bug in fedora template installation with dispvm service VMs and sys-net. Closed but not fixed. Testers confused. Testing processes unclear. Need discussion on forum.

Possible solution: Add an installer issue tag that will trigger template rebuilds when a package that fixes an installer issue is built, so that the next openqa/weekly builds create ISOs that include the templates containing the fix?

**Comment by marmarek** [comment]
> Does this problem impact anyone whose ISO includes [QubesOS/qubes-core-agent-linux#437](https://github.com/QubesOS/qubes-core-agent-linux/pull/437)?

Probably no, but nobody tried because such ISO does not exist (yet). Fedora templates (until yesterday) were built before the fix was pushed, so any ISO did not include it. That's the whole confusion @tlaurion is talking about.
I think the simple improvement to the process would be to build templates more often (every week? every two weeks?).

**Comment by tlaurion** [comment]
> > Does this problem impact anyone whose ISO includes [QubesOS/qubes-core-agent-linux#437](https://github.com/QubesOS/qubes-core-agent-linux/pull/437)?
> 
> Probably no, but nobody tried because such ISO does not exist (yet). Fedora templates (until yesterday) were built before the fix was pushed, so any ISO did not include it. That's the whole confusion @tlaurion is talking about.
> I think the simple improvement to the process would be to build templates more often (every week? every two weeks?).

@marmarek I would really love to see
- issues tagged for "installer issue" 
- pr fixing the issue being trackable in built package (downloadable as testing package from repo) commenting/referencing back to issue 
- templates including fixed packages being built  commenting/referencing back to issue
- iso being built with updated templates commenting/referencing back to issue


Sorry for my walls of text @andrewdavidwong but if we want to test fixes, we need to know which package version includes the fix, we need to know which template version includes the fixed packages and we need to know which ISO fixes known installer issues. Even more for a fix for installer.

----

Some history unrelated to this issue:
- q4.0 changed pool config of TLVM, separating dom0 from VM pools. That required a reinstallation
- randomization of mac addresses required a reinstallation of template/sys-net servicevm, otherwise testers don't know, issues were opened, documentation was off and still today there is confusion on what needs to be done to anonymize properly
- q4.0 upgrade script attempted to deal with pool changes giving warning, but users thought it would work out of the box where reinstallation was really the only solution.
- LUKs v2 happened between 3.2 and 4.0. This is an important "installer" change that create a lot of rockus.
- down to drive TRIM, LUKs discards, LVM discards, VM fs discard. Heads applying discard on crypttab file generation injected in secret.cpio to do apply disk unlock key from TPM unsealed disk encryption key passphrase.

Nothing "installer" related problems can easily fix those issues but to backup/reinstall/restore.

Some thoughts.

Installer "tag" should encompass:
- pool creation changes
- filesystem creation assumption changes
- templates upgrade for networking
- dom0 changes in filesystem management
- dom0 tooling affecting filesystem optimizations
- anything else that cannot effectively be retroactively fixed
- installer issues to be bootable by iso
- installer issue to be installable on iso origin drive but different partition (optimal tester scenario)

----

Other examples

A recent bug report to NixOS made them realize that LUKs unlocking was insecure on legacy firmware and that unlocking secret was stored unencrypted on /boot.... Reencrypting LUKS online was an option. Reinstalling is the solution. 

Pool change seperation of q4.0 required a reinstall. Upgrade scripts lowered security. Let's remember that LUKS1 was effective prior or that. Reencrypting is possible to upgrade to LUKS2 but far from being recommended. users should reinstall in all case (otherwise unclear why they use Quebesos for insecure use case)

----

Tldr: an installer issue tag made available globally would track reinstallation requirements and track needed testing more efficiently.

**Comment by samtinel** [comment]
I don't fully understand if feedback is still asked for here, but updating qubes-core-agent to testing fixed it for me.

**Comment by andrewdavidwong** [comment]
This issue has been closed as resolved. If anyone believes this issue is not yet resolved, or if anyone is still affected by this issue, please leave a comment, and we'll be happy to reopen it. Thank you.
